File: src/my_project/easy_problems/from151To200/determineStringHalves.py
from typing import List 
from collections import defaultdict

class Solution:
    def halvesAreAlike(self, s: str) -> bool:
        left_s = s[0:len(s)//2]
        right_s = s[len(s)//2:]
        count_left: int = 0
        count_right: int = 0
        

        vowels =  ['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U']

        for i in range(len(s)//2):
            if left_s[i] in vowels:
                count_left += 1
            if right_s[i] in vowels:
                count_right += 1

        if count_left != count_right:
            return False 
        else: 
            return True 



solution = Solution()

print(solution.halvesAreAlike(s = "book"))

print(solution.halvesAreAlike(s = "textbook"))

print(solution.halvesAreAlike(s = "MerryChristmas"))

print(solution.halvesAreAlike(s = "AbCdEfGh"))

File: src/my_project_typescript/easy_problems_typescript/from51To100/determineStringHalves.ts
function halvesAreAlike(s: string): boolean {
    let leftS: string = s.slice(0, Math.floor(s.length /2))
    let rightS: string = s.slice(Math.floor(s.length /2))
    let countLeft: number = 0
    let countRight: number = 0

    let vowels: Array<string> =  ['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U']

    for (let i = 0; i < Math.floor(s.length /2); i++){
        if (vowels.includes(leftS[i])){
            countLeft += 1
        }
        if (vowels.includes(rightS[i])){
            countRight += 1
        }
    }
    if (countLeft !== countRight){
        return false 
    }
    else{
        return true 
    }
    

};


console.log(halvesAreAlike( "book"))

console.log(halvesAreAlike("textbook"))

console.log(halvesAreAlike("MerryChristmas"))

console.log(halvesAreAlike("AbCdEfGh"))
